core/src/main/scala/chisel3/experimental/hierarchy/ModuleClone.scala

```diff
@@ -30,15 +30,16 @@ private[chisel3] class ModuleClone[T <: BaseModule](val getProto: T)(implicit si
     _component = getProto._component
     None
   }
-  // Maps proto ports to module clone's ports
+  // Maps proto ports to module clone's ports.
+  // Chisel ports can be Data or Property, but to clone as a Record, we can only return Data.
   private[chisel3] lazy val ioMap: Map[Data, Data] = {
     getProto match {
       // BlackBox needs special handling for its pseduo-io Bundle
       case protoBB: BlackBox =>
         Map(protoBB._io.get -> getPorts._elements("io"))
       case _ =>
         val name2Port = getPorts._elements
-        getProto.getChiselPorts.map { case (name, data) => data -> name2Port(name) }.toMap
+        getProto.getChiselPorts.collect { case (name, data: Data) => data -> name2Port(name) }.toMap
```

Comment:
This is the same thing about cloning Modules, this is only cloning their Data ios.